Neos.Flow/Classes/Mvc/Routing/RouteConfiguration.php

```diff
@@ -0,0 +1,120 @@
+    ) {
+    }
+
+    /**
+     * Returns a list of configured routes
+     *
+     * @return Routes
+     */
+    public function getRoutes(): Routes
+    {
+        if ($this->routesCreated === false) {
+            $this->createRoutesFromConfiguration();
+        }
+        return $this->routes;
+    }
+
+    /**
+     * Creates \Neos\Flow\Mvc\Routing\Route objects from the injected routes
+     * configuration.
+     *
+     * @return void
+     * @throws InvalidRouteSetupException
+     */
+    protected function createRoutesFromConfiguration(): void
+    {
+        $routesConfiguration = $this->configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_ROUTES);
+        $routes = [];
+        $routesWithHttpMethodConstraints = [];
+        foreach ($routesConfiguration as $routeConfiguration) {
+            $route = new Route();
+            if (isset($routeConfiguration['name'])) {
+                $route->setName($routeConfiguration['name']);
+            }
+            $uriPattern = $routeConfiguration['uriPattern'];
+            $route->setUriPattern($uriPattern);
+            if (isset($routeConfiguration['defaults'])) {
+                $route->setDefaults($routeConfiguration['defaults']);
+            }
+            if (isset($routeConfiguration['routeParts'])) {
+                $route->setRoutePartsConfiguration($routeConfiguration['routeParts']);
+            }
+            if (isset($routeConfiguration['toLowerCase'])) {
+                $route->setLowerCase($routeConfiguration['toLowerCase']);
+            }
+            if (isset($routeConfiguration['appendExceedingArguments'])) {
+                $route->setAppendExceedingArguments($routeConfiguration['appendExceedingArguments']);
+            }
+            if (isset($routeConfiguration['httpMethods'])) {
+                if (isset($routesWithHttpMethodConstraints[$uriPattern]) && $routesWithHttpMethodConstraints[$uriPattern] === false) {
+                    throw new InvalidRouteSetupException(sprintf('There are multiple routes with the uriPattern "%s" and "httpMethods" option set. Please specify accepted HTTP methods for all of these, or adjust the uriPattern', $uriPattern), 1365678427);
+                }
+                $routesWithHttpMethodConstraints[$uriPattern] = true;
+                $route->setHttpMethods($routeConfiguration['httpMethods']);
+            } else {
+                if (isset($routesWithHttpMethodConstraints[$uriPattern]) && $routesWithHttpMethodConstraints[$uriPattern] === true) {
+                    throw new InvalidRouteSetupException(sprintf('There are multiple routes with the uriPattern "%s" and "httpMethods" option set. Please specify accepted HTTP methods for all of these, or adjust the uriPattern', $uriPattern), 1365678432);
+                }
+                $routesWithHttpMethodConstraints[$uriPattern] = false;
+            }
+            $routes[] = $route;
```

Comment:
What about moving all that logic to some 
```php
Routes::fromConfiguration(array $configuration)
```
method that calls some
```php
Route::fromConfiguration(array $configuration)
```
method internally?

This way, this class turns into a really simple layer between configuration manager & router